src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -479,6 +491,8 @@ public void setField(Map<String, String> fields) {
 
         fields.remove(fieldName);
         fieldsAsWords.remove(fieldName);
```

Comment:
maybe also move `fieldsAsWords.remove(fieldName);` to the invalidate cache method?